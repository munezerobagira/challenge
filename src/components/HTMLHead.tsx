import React from 'react';
import Head from 'next/head';
export interface HTMLHeadProps {
  description: string;
  title: string;
}
function HTMLHead({ description, title }: HTMLHeadProps) {
  return (
    <Head>
      <title>{title}</title>
      <meta name="description" content={description} data-testid="html-description" />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <link rel="icon" href="/favicon.ico" />
    </Head>
  );
}

export default HTMLHead;
