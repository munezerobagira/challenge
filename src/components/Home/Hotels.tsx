import hotel from '../../interfaces/hotel';
import hotelStyles from './hotels.module.scss';
import Card from './Card/Card';
import Link from 'next/link';

export interface HotelsProps {
  hotels: (hotel | null)[];
}
function Hotels({ hotels }: HotelsProps) {
  return (
    <div className={`${hotelStyles.hotels}`}>
      {hotels.map((hotel, key) => (
        <Card hotel={hotel} key={hotel?.id || key} />
      ))}
    </div>
  );
}

export default Hotels;
