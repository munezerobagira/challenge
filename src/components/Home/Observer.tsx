import React, { RefObject } from 'react';
interface ObserverObject extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
  target: React.LegacyRef<HTMLDivElement>;
  height?: number | string;
  width?: number | string;
}

function Observer({ target, height = 400, width = '100%', ...resProps }: ObserverObject) {
  return <div ref={target} style={{ height, width }} {...resProps}></div>;
}

export default Observer;
