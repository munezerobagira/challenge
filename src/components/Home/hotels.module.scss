.hotels {
  @apply container grid  gap-4 py-4 grid-cols-1 sm:grid-cols-2 md:grid-cols-4;
}
