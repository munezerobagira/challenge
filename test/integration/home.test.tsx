import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HTMLHead, { HTMLHeadProps } from '../../src/components/Common/HTMLHead';
import Home from '../../src/pages';

describe('Home page', () => {
  it('render link to profile the about page', () => {
    const headElement = renderer.create(<Home />);
    expect(headElement.toJSON()).toMatchSnapshot();
  });
});
