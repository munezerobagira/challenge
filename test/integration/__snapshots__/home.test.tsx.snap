// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render link to profile the about page 1`] = `null`;
